### DISCUSSION TITLE
Shouldn't embedded:io/SMBus extend embedded:io/I2C?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#1151] [Q&A]
In ECMA419 I read "The SMBus class extends the I2C class with additional methods" yet calling `write()` on an SMBus object produces a "no function!" exception and looking at smbus.js I don't see an implementation of write nor extension of I2C. Am I missing something?

The use-case is a TI power management chip that has a PMBus interface, which is an extension of SMBus, and it has "commands" where one just writes the command/register byte and no data. It looks like there's a non-standard `sendByte()` function that should do what I need...

**Comment by phoddie** [comment]
Yes, our ECMA-419 SMBus implementation should provide the `read` and `write` methods from I2C.  That's an oversight and should be corrected.

> It looks like there's a non-standard sendByte() function that should do what I need...

`sendByte()` is standard in ECMA-419 2nd Edition, which is scheduled for review and approval (I hope) at the June GA later this month. Here's the [relevant section](https://github.com/EcmaTC53/spec/blob/master/docs/tc53.md#methods-2).
